core/lib/dal/src/blocks_dal.rs

```diff
@@ -558,8 +558,8 @@ impl BlocksDal<'_, '_> {
             initial_bootloader_contents,
             used_contract_hashes,
             base_fee_per_gas,
-            header.l1_gas_price as i64,
-            header.l2_fair_gas_price as i64,
+            (header.l1_gas_price).as_u64() as i64,
+            (header.l2_fair_gas_price).as_u64() as i64, // TODO: this might overflow
```

Comment:
Fixing this overflow requires a database migration moving the field from `bigint` into `numeric(80,0)`. What is the policy for postgres migrations? Do we change the existing field or should we create a new one to start using now?

Reply:
I added a migration creating new columns for the values that need to be moved to U256 here [518ee35](https://github.com/matter-labs/zksync-era/pull/1324/commits/518ee352e221b836fb5311b7962c2d4db5a146a4). I don't believe it's possible to have an old server version work with a DB with applied migrations if we simply modify the column type.

Essentially the migration creates new columns with a `_u256` suffix, copies all the existing values on the non-`_u256` column over to it, and the rust `dal`/`sqlx` code is updated to make selects take the `_u256` version and map it to the existing struct fields. Inserts are updated to insert into the new `_u256` column, and an `SQL` trigger is created for inserts and updates to those tables to also insert/update the old column with the same value. The only issue here is what happens when the value inserted/updated does not fit in a `bigint`/`i64`. The trigger implementation I pushed will set the old column value to `-1` in that case to signify an error. 

Let me know what you think of this, in theory that scenario should not be a problem, as in the currently deployed environments the value should always fit in a `bigint`, but it might be better to error out in those environments to make absolutely nothing weird happens.